File: Domashno_2/guessable-flutter-client/guessable/lib/domain/guess.dart
import 'package:equatable/equatable.dart';
import 'package:guessable/domain/location.dart';

/// Represents a guess for a location for the application
///
/// * [id] - the unique identifier for the [Guess]
/// * [location] - the [Location] this guess is meant for
/// * [guessedLatitude] - the guessed map latitude for the given location
/// * [guessedLongitude] - the guessed map longitude for the given location
/// * [distance] - the distance between the guess and the actual location in kilometers
/// * [Guess.fromJson] - used for creating a [Guess] object from JSON, typically from HTTP requests
class Guess extends Equatable {
  final int id;
  final Location location;
  final double guessedLatitude;
  final double guessedLongitude;
  final double distance;

  const Guess(
      {required this.id,
      required this.location,
      required this.guessedLatitude,
      required this.guessedLongitude,
      required this.distance});

  factory Guess.fromJson(Map<String, dynamic> json) {
    return Guess(
        id: json['id'],
        location: Location.fromJson(json['location']),
        guessedLatitude: json['guessedLatitude'],
        guessedLongitude: json['guessedLongitude'],
        distance: json['distance']);
  }

  @override
  List<Object?> get props => [id, location, guessedLatitude, guessedLongitude, distance];
}

File: Domashno_2/guessable-api/src/main/kotlin/mk/ukim/finki/mis/guessableapi/domain/Guess.kt
package mk.ukim.finki.mis.guessableapi.domain

import jakarta.persistence.*


/**
 * Represents a guess for an existing [Location]
 *
 * @property[id] unique identifier for the guess
 * @property[location] the [Location] to which this guess is meant for
 * @property[guessedBy] the user that created the guess
 * @property[guessedLatitude] the map latitude of the guess
 * @property[guessedLongitude] the map longitude of the guess
 * @property[distance] the distance from the guess to the actual location in kilometers
 * */
@Entity
@Table(name = "guesses")
data class Guess(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long,

    @ManyToOne
    @JoinColumn(name = "location")
    val location: Location,

    @ManyToOne
    @JoinColumn(name = "guessed_by")
    val guessedBy: GuessableUser,

    @Column(name = "guessed_latitude")
    val guessedLatitude: Double,

    @Column(name = "guessed_longitude")
    val guessedLongitude: Double,

    @Column(name = "distance")
    val distance: Double,
)
